### PR TITLE
chore: Replace `black` by `ruff format`

### PR DESCRIPTION
**Pull request by stinodego**
Closes https://github.com/pola-rs/polars/issues/11993

Overal, some nice formatting improvements, and of course, much faster formatting!

### CODE REVIEW

File: py-polars/tests/unit/series/test_series.py [hunk 1]
```diff
@@ -431,7 +431,7 @@ def test_power() -> None:
     assert_series_equal(a**a, pl.Series([1.0, 4.0], dtype=Float64))
     assert_series_equal(b**b, pl.Series([None, 4.0], dtype=Float64))
     assert_series_equal(a**b, pl.Series([None, 4.0], dtype=Float64))
-    assert_series_equal(a**None, pl.Series([None] * len(a), dtype=Float64))
+    assert_series_equal(a ** None, pl.Series([None] * len(a), dtype=Float64))
```
Comment (stinodego) [inline py-polars/tests/unit/series/test_series.py:434]: This seems to be an error in the Ruff formatter, will open an issue for it.

File: py-polars/polars/dataframe/frame.py [hunk 2]
```diff
@@ -174,10 +175,10 @@
...
     # NOTE: wrapping these as strings is necessary for Python <3.10
 
-    MultiRowSelector: TypeAlias = "slice | range | list[int] | Series"
-    MultiColSelector: TypeAlias = (
-        "slice | range | list[int] | list[str] | list[bool] | Series"
-    )
+    MultiRowSelector: TypeAlias = Union[slice, range, list[int], "Series"]
+    MultiColSelector: TypeAlias = Union[
+        slice, range, list[int], list[str], list[bool], "Series"
+    ]
```
Comment (stinodego) [inline py-polars/polars/dataframe/frame.py:181]: ruff had some trouble formatting these strings nicely, so I replaced them with Unions.

File: py-polars/tests/unit/dataframe/test_df.py [hunk 3]
```diff
@@ -2774,15 +2772,17 @@ def test_selection_regex_and_multicol() -> None:
...
-            "b": [2, 3],
-        }
+
+    result = df.with_columns([pl.col("a").set_sorted()]).unique(
+        subset=subset,
+        keep="last",
+    )
+
+    expected = pl.DataFrame({"a": [1, 3], "b": [2, 3]})
+    assert_frame_equal(result, expected)
```
Comment (stinodego) [inline py-polars/tests/unit/dataframe/test_df.py:2785]: I did some additional rewriting of this test myself, as the new formatting had misplaced the type ignore comment.

File: py-polars/polars/dataframe/frame.py [hunk 4]
```diff
@@ -174,10 +175,10 @@
...
     # NOTE: wrapping these as strings is necessary for Python <3.10
 
-    MultiRowSelector: TypeAlias = "slice | range | list[int] | Series"
-    MultiColSelector: TypeAlias = (
-        "slice | range | list[int] | list[str] | list[bool] | Series"
-    )
+    MultiRowSelector: TypeAlias = Union[slice, range, "list[int]", "Series"]
+    MultiColSelector: TypeAlias = Union[
+        slice, range, "list[int]", "list[str]", "list[bool]", "Series"
+    ]
```
Comment (stinodego) [inline py-polars/polars/dataframe/frame.py:181]: ruff had some trouble formatting these strings nicely, so I replaced them with Unions.

**Comment by ritchie46** [comment]
Alright, shall we wait a little bit to get all issues ironed out?

**Comment by stinodego** [comment]
> Alright, shall we wait a little bit to get all issues ironed out?

I don't think that's necessary. They are very minor issues and the workarounds are fine. And the formatter has improved things in a lot of other places!

**Comment by alexander-beedie** [comment]
Happy days... I switched a chunk of the work codebase onto `ruff format` this week - so far so good! ✌️

**Pull request merged**